lib/sparrow/h2_worker/pool.ex

```diff
@@ -87,7 +87,8 @@ defmodule Sparrow.H2Worker.Pool do
   @spec start_link(Sparrow.H2Worker.Pool.Config.t(), pool_type, [atom]) ::
           {:error, any} | {:ok, pid}
   def start_link(config, pool_type, tags \\ []) do
-    Sparrow.PoolsWarden.add_new_pool(pool_type, config.pool_name, tags)
-    start_link(config)
+    {:ok, pid} = start_link(config)
+    Sparrow.PoolsWarden.add_new_pool(pid, pool_type, config.pool_name, tags)
```

Comment:
If there is no `pool_name` in config, than pool will start but fail to register in `PoolsWarden`, right?
how about:
```elixir
pool_name = config.pool_name
{:ok, pid} = start_link(config)
Sparrow.PoolsWarden.add_new_pool(pid, pool_type, pool_name, tags)
```
Than if there is no `pool_name` in config it will fail before starting the pool.
Correct me if I am wrong.